crates/runtime/src/user_agent_util.rs

```diff
@@ -0,0 +1,40 @@
+use regex::Regex;
+use std::sync::LazyLock;
+
+static UA_REGEX: LazyLock<Result<Regex, regex::Error>> =
+    LazyLock::new(|| Regex::new(r"^(\S+) (\S+) \((.*?)\)$"));
```

Comment:
```suggestion
    LazyLock::new(|| Regex::new(r"^(spice-rs|spice-java|spice.js|spicepy|gospice) v?(\d\.?)+(\-\w+)? \((.*?)\)$"));
```

could we use a stricter regex here? suggestion isn't fully fleshed, but e.g. specifying the specific values for each SDK user-agent value, checking version string format, etc.

Reply:
I guess we'd need to solve for the capture groups differently if we make the regex stricter though